flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/MySqlChunkSplitter.java

```diff
@@ -220,7 +220,17 @@ private List<ChunkRange> splitEvenlySizedChunks(
         while (ObjectUtils.compare(chunkEnd, max) <= 0) {
             splits.add(ChunkRange.of(chunkStart, chunkEnd));
             chunkStart = chunkEnd;
-            chunkEnd = ObjectUtils.plus(chunkEnd, chunkSize);
+            try {
+                chunkEnd = ObjectUtils.plus(chunkEnd, chunkSize);
+            } catch (ArithmeticException e) {
+                // Set chunkEnd to the max value of number type to avoid dead loop if number
+                // overflows.
+                chunkEnd = chunkEnd instanceof Integer ? Integer.MAX_VALUE : Long.MAX_VALUE;
+            }
+            // If chunkStart equals chunkEnd,should stop splitting chunk.
+            if (ObjectUtils.compare(chunkStart, chunkEnd) == 0) {
+                break;
+            }
```

Comment:
Could we jump out of the loop here directly in the catch clause? Then all records greater than `chunkStart` (`chunkStart` to `MAX` in overflow case) will be treated as the ending split.

Reply:
@PatrickRen I have fixed.Thks for your tips.